File: system-tests/selenium-js/driverFactory.js
const os = require('os');
const { Builder } = require('selenium-webdriver');
const chrome = require('selenium-webdriver/chrome');

function buildDriver() {
  const options = new chrome.Options();
  options.setUserPreferences({
    'profile.password_manager_leak_detection': false,
    'credentials_enable_service': false,
    'profile.password_manager_enabled': false,
  });

  // Headless solo en Linux (Ubuntu Server no tiene interfaz gráfica)
  if (os.platform() === 'linux') {
    options.addArguments('--headless=new');
    options.addArguments('--no-sandbox');
    options.addArguments('--disable-dev-shm-usage');
  }

  return new Builder().forBrowser('chrome').setChromeOptions(options).build();
}

module.exports = { buildDriver };

File: system-tests/selenium-java/src/test/java/com/qatraining/DriverFactory.java
package com.qatraining;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.chrome.ChromeOptions;

import java.time.Duration;
import java.util.HashMap;
import java.util.Map;

public class DriverFactory {

    public static WebDriver createDriver() {
        ChromeOptions options = new ChromeOptions();

        Map<String, Object> prefs = new HashMap<>();
        prefs.put("profile.password_manager_leak_detection", false);
        prefs.put("credentials_enable_service", false);
        prefs.put("profile.password_manager_enabled", false);
        options.setExperimentalOption("prefs", prefs);

        // Headless solo en Linux (Ubuntu Server no tiene interfaz gráfica)
        String os = System.getProperty("os.name").toLowerCase();
        if (os.contains("nux")) {
            options.addArguments("--headless=new");
            options.addArguments("--no-sandbox");
            options.addArguments("--disable-dev-shm-usage");
        }

        WebDriver driver = new ChromeDriver(options);
        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(5));
        return driver;
    }
}